### PR TITLE
Add parser

### PR DESCRIPTION
**Pull request by meta-closure**
add extra data parser and dump command

- Parse

Can parse extra data, go_type, column_name, column_type. for example 

``` pass.yml;yaml

  test_multitype:
    title: test multitype
    type: integer
    go_type: int
    column:
      name: test multitype
      db_type: int
      foreign_key:
        name: fk

```

this JSON Schema(yaml) file convert into a inter-generate go-jstmpl data structure
such that...

```result.yml;yaml

- ColumnName: test multitype
  ColumnType: int
  GoType: int
  Name: TestMultitype
  Type: number

```

this is good for writing text template, make easy

- Dump

add dump command, it is required a file path, or stdout specification

### CODE REVIEW

File: builder_test.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+	if len(ts.Definitions) != 7 {
+		t.Fatal("fail to parse Definitions")
+	}
+
+	for _, v := range ts.Properties {
+		switch v.Key() {
+		case "test_multitype":
+			i, ok := v.(*jstypes.Integer)
+			if ok != true {
```
Comment (minodisk) [inline builder_test.go]: @meta-closure 

You should shorten `ok != true` to `!ok`.
I found 11 same cases in this file.

File: builder.go [hunk 2]
```diff
@@ -25,12 +25,14 @@ func (b *Builder) Build(hs *hschema.HyperSchema) (*types.Root, error) {
...
+	if hs.Schema.Extras["href"] != nil {
+		str := hs.Schema.Extras["href"].(string)
 
+		m.URL, err = url.Parse(str)
+		if err != nil {
+			return nil, err
+		}
+	}
 	ctx := &types.Context{
 		Validations: map[string]bool{},
```
Comment (minodisk) [inline builder.go]: `RawSchema: s,`

**Comment by meta-closure** [comment]
Fix done!

File: test/combine.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  test_object:
+    title: test object
+    type: object
+    properties:
+        conbinatrial:
+            $ref: '#/definitions/test_integer'
+            column:
+                name: test_parts
+                db_type: test_db_type
```
Comment (minodisk) [inline test/combine.yml]: @meta-closure Add newline to the end of this file.

**Comment by meta-closure** [comment]
@minodisk add!

**Pull request merged**